File: Backend/JAVA/application/src/main/java/com/superhero/domain/model/custom_model/CompleteHero.java
package com.superhero.domain.model.custom_model;

import com.fasterxml.jackson.annotation.JsonGetter;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonSetter;
import com.superhero.domain.model.Appearance;
import com.superhero.domain.model.Biography;
import com.superhero.domain.model.Connections;
import com.superhero.domain.model.Image;
import com.superhero.domain.model.Powerstats;
import com.superhero.domain.model.Work;

public class CompleteHero {
    
    @JsonProperty("response")
    public String Response;

    @JsonProperty("powerstats")
    public Powerstats Powerstats;
    
    @JsonProperty("biography")
    public Biography Biography;

    @JsonProperty("appearance")
    public Appearance Appearance;
    
    @JsonProperty("work")
    public Work Work;
    
    @JsonProperty("connections")
    public Connections Connections;
    
    @JsonProperty("image")
    public Image Image;

    @JsonProperty("name")
    public String Name;

    @JsonProperty("id")
    public String PublicId;

    public CompleteHero() {
    }

    public String getResponse() {
        return Response;
    }

    public void setResponse(String response) {
        Response = response;
    }

    public Powerstats getPowerstats() {
        return Powerstats;
    }

    public void setPowerstats(Powerstats powerstats) {
        Powerstats = powerstats;
    }
    
    public Biography getBiography() {
        return Biography;
    }

    public void setBiography(Biography biography) {
        Biography = biography;
    }
    
    public Appearance getAppearance() {
        return Appearance;
    }

    public void setAppearance(Appearance appearance) {
        Appearance = appearance;
    }
    
    public Work getWork() {
        return Work;
    }

    public void setWork(Work work) {
        Work = work;
    }
    
    public Connections getConnections() {
        return Connections;
    }

    public void setConnections(Connections connections) {
        Connections = connections;
    }
    
    public Image getImage() {
        return Image;
    }

    public void setImage(Image image) {
        Image = image;
    }
    
    public String getName() {
        return Name;
    }

    public void setName(String name) {
        Name = name;
    }
    
    public String getPublicId() {
        return PublicId;
    }

    public void setPublicId(String publicId) {
        PublicId = publicId;
    }
}

File: Backend/Typescript/src/domain/entities/custom/complete-hero.ts
import { Expose } from "class-transformer";
import { Powerstats } from "../Powerstats.js";
import { Biography } from "../Biography.js";
import { Appearance } from "../Appearance.js";
import { Work } from "../Work.js";
import { Connections } from "../Connections.js";
import { Image } from "../Image.js";
import { GuidId } from "../base/guid-id.js";

export class CompleteHero extends GuidId {
    @Expose({ name: "powerstats" })
    powerstats: Powerstats = new Powerstats();

    @Expose({ name: "biography" })
    biography: Biography = new Biography();

    @Expose({ name: "appearance" })
    appearance: Appearance = new Appearance();

    @Expose({ name: "work" })
    work: Work = new Work();

    @Expose({ name: "connections" })
    connections: Connections = new Connections();

    @Expose({ name: "image" })
    image: Image = new Image();

    @Expose({ name: "name" })
    name: string | undefined;

    @Expose({ name: "public_id" })
    PublicId: string | undefined;
}